prime-router/src/main/kotlin/Report.kt

```diff
@@ -287,7 +288,7 @@ class Report : Logging {
         this.bodyFormat = bodyFormat ?: destination?.format ?: Format.INTERNAL
         this.itemLineages = itemLineage
         this.createdDateTime = OffsetDateTime.now()
-        this.metadata = Metadata.getInstance()
+        this.metadata = metadata ?: Metadata.getInstance()
```

Comment:
For a follow up ticket: The `Report` class doesn't actually need metadata. It looks like it's only used by some fake data generation which should be a separate concern. Extracting that behavior would simplify things.